### ISSUE TITLE
Download all .json files from Tabular Editor 

### ISSUE THREAD

**Issue by lakviat** [TabularEditor/TabularEditor#991]
Hi there.

I'm trying to download all .json measures from the Tabular Editor that reads from power bi (.pbix file)
I need to integrate this with CLI so that I can run in CI/CD

The previous issues raised here didn't work for me ( I did researches before raising issue)
'''
$cmd = "& ""C:\Program Files (x86)\Tabular Editor\TabularEditor.exe"" ""$Server"" ""$DatabaseName"" -F ""$ModelFolderPath""  "
'''

This is the command I came up with : 
- - - - 
start "& ""C:\Program Files\Tabular Editor 3\TabularEditor3.exe"" ""$https://dbname.com"" ""$ENTERPRISE_DB""  -F ""$C:\Users\Test""  "
- - - -

Basically I need to extract all measure .json files from the Tabular Editor from the CLI but none of the options worked for me 

I would appreciate the help !

**Comment by otykier** [comment]
Hi @lakviat 
In your example, you're using TabularEditor3.exe. This executable does not have a CLI. You should use the .exe from [Tabular Editor 2.x](https://github.com/TabularEditor/TabularEditor/releases/tag/2.16.6) instead.

**Comment by lakviat** [comment]
Thanks getting back to me @otykier 

I tried with the version 2 but that didn't work for me, am I missing something ? 

command that I'm running : 
- - - - - - - - - 
start /wait "& ""C:\Program Files (x86)\Tabular Editor\TabularEditor.exe"" ""$powerbi://api.powerbi.com"" ""$DB_ID_Number"" -F ""$C:\Users\user_name\Folder""  "

- - - - - - - - -
I would appreciate the help here

**Comment by otykier** [comment]
What error message are you getting?

**Comment by lakviat** [comment]
@otykier Basically nothing, it opens another CMD terminal and that's it

**Comment by otykier** [comment]
Okay, I suggest you try to run the command on your local machine in a Windows Command Prompt, to better see what’s going on, before implementing it in a CI/CD pipeline.

**Comment by otykier** [comment]
Try to run it without `start /wait`. It’s difficult to know whats going on if you’re not getting any output.

**Comment by lakviat** [comment]
@otykier now I'm getting the message 

-- -- -- -- -- 
The filename, directory name, or volume label syntax is incorrect (The directory is present thought)
-- -- -- -- -- 

The command I ran 

-- -- -- -- -- 
"& ""C:\Program Files (x86)\Tabular Editor\TabularEditor.exe"" ""$powerbi://api.powerbi.com"" ""$Labour Dev"" -F ""$C:\Users\nmirovich\BigTEst"" "

-- -- -- -- --

**Comment by otykier** [comment]
That error message does not originate from Tabular Editor. Let's start over. Here's what I do:

```cmd
"c:\Program Files (x86)\Tabular Editor\TabularEditor.exe" "localhost" "AdventureWorks" -F "c:\Users\DanielOtykier\Desktop\TestOutputModel"
```

Here, you would need to replace `localhost` with the full connection string to the Power BI workspace, **including** your Service Principal credentials, there's an [example here](https://powerbi.microsoft.com/en-us/blog/announcing-support-for-service-principals-using-the-xmla-endpoint-in-power-bi-premium/). `AdventureWorks` is the name of the dataset to load. You should see something like this to indicate that TabularEditor.exe is actually running:

![image](https://user-images.githubusercontent.com/8976200/169465172-9bcfed95-dc4f-4fec-8353-ebb67b4f1042.png)

**Comment by lakviat** [comment]
Hi @otykier thanks getting back to me 

I created this Connection String in Power Shell 

![image](https://user-images.githubusercontent.com/29146176/169843838-c717fbd8-d677-4d6e-b988-066cb80220ed.png)


And ran command like 

```
"C:\Program Files (x86)\Tabular Editor\TabularEditor.exe" "ConnectionString.ps" "Labour DEV" -F "C:\Users\nmirovich\BigTest"
```
The output I'm getting 

![image](https://user-images.githubusercontent.com/29146176/169844174-38e5c3da-49b0-4ffb-a35f-8d4b315bd16f.png)

A connection cannot be made. Ensure that the server is running 
Basically I have Tabular Editor running in my local with the Power Bi but I'm not sure why I'm not getting connected

**Comment by lakviat** [comment]
Thanks for the help @otykier 

Just to let know "Provider=MSOLAP; was missing from the command for that reason it was not generating folders 

works now

**Comment by lakviat** [comment]
Hi @otykier quick question, have you solved the problem of programmatically downloading the .pbix files from power bi to your CI CD server by any chance ? if so can you share some insights that worked best for you ?
